### ISSUE TITLE
500 error authenticating with AirPlay devices using a home password

### ISSUE THREAD

**Issue by ohmantics** [owntone/owntone-server#1527]
There are mentions of this in at least two bug reports here, but I figured it best to break it into it's own bug as I think I've narrowed it down.

If you have a password set in Home Settings->Allow Speaker & TV Access (Require Password is enabled), then all attempts to enable an affected device in owntone results in a 500 error from the device. Disabling Require Password works properly. This affects HomePods, Airport Express, etc. as auth is handled differently from AppleTV and macOS devices.

Setting an individual device password in owntone.conf doesn't seem to work for me for this situation.

**Comment by ejurgensen** [comment]
Yes, perhaps I should also add a note to the docs that password auth currently isn't supported for Airplay 2.

**Comment by ohmantics** [comment]
Do we know how to handle this sort of authentication?

**Comment by Porco-Rosso** [comment]
There have been some advances in https://github.com/openairplay/airplay2-receiver
Maybe they have solved it?

**Comment by ejurgensen** [comment]
I haven't really looked at AP2 password based auth. I don't recall having seen any info on how it works. Does airplay2-receiver support password based auth, @Porco-Rosso?

**Comment by Porco-Rosso** [comment]
I don't know, I haven't had a chance to try it out yet. I assumed "HomeKit non-transient pairing" might include password auth. There is some discussion on password auth here: https://github.com/openairplay/airplay2-receiver/pull/41

**Comment by ejurgensen** [comment]
Non-transient pairing is PIN based, which OwnTone supports. I did try airplay2-receiver some time ago, and I don't think it had pwd auth back then.

**Comment by ejurgensen** [comment]
I see one of the comments mentions pwd and says the client just sends it in plain with POST /configure. If that is correct it would be really easy, so I will try that.

**Comment by ejurgensen** [comment]
The "POST /configure" method didn't pan out, but while testing I could see that simply sending a password instead of an OTP seemed to work. So know I've added experimental support for password based auth in the above branch (ap2_password_auth1). I have tested both with password set on speaker (no HomeKit), and with password set via the Home app.

The password is provided to OwnTone via the config file, so:

```
airplay "SpeakerName" {
 password = "123456"
}
```
It would be great if you can test the change.

**Comment by simonsolts-bark** [comment]
@ejurgensen Hi, and thanks for investigating this issue - I can't tell if it's been merged into main or not? I am experiencing exactly the same issue: HomePod speaker works fine with "Anyone on same network | require password: OFF", but fails with require password ON.

Attempted to add the speaker manually to my config file with the password and force roap off (i.e. force airplay2) but no joy. 

Think this is relevant from the logs: 
```[2022-10-11 15:44:14] [  LOG]  airplay: Response to SETUP (session) from 'Simon’s Room' was negative, aborting (401 Unauthorized)
[2022-10-11 15:44:14] [  LOG]  airplay: Clearing 'Simon’s Room' pairing keys, you need to pair again
[2022-10-11 15:44:14] [  LOG]   player: The AirPlay 2 device 'Simon’s Room' failed to activate
[2022-10-11 15:44:14] [  LOG]      web: JSON api request failed with error code 500 (/api/outputs/163733592531654/toggle)
[2022-10-11 15:44:16] [  LOG]  airplay: Pair setup stage complete, saving authorization key
[2022-10-11 15:44:18] [  LOG]  airplay: Response to SETUP (session) from 'Simon’s Room' was negative, aborting (401 Unauthorized)
[2022-10-11 15:44:18] [  LOG]  airplay: Clearing 'Simon’s Room' pairing keys, you need to pair again
[2022-10-11 15:44:18] [  LOG]   player: The AirPlay 2 device 'Simon’s Room' failed to activate```


If your change has been merged in (I am on latest release: Version 28.5) then maybe I am doing something wrong; if not; is there a branch I can test on? Couldn't find the branch you referred to. 

Thanks in advance!

**Comment by ejurgensen** [comment]
It's merged and included in 28.5, so it should work. From the logs it seems pairing is going ok with the password, but then the Homepod replies with 401 to the subsequent session start request. If you try to give OwnTone an incorrect password I expect you will see the pairing fails and the 401 is never reached - can you confirm that is the case? Also, please create a new issue and attach with the same part of the log, just with the log level set to debug.

**Comment by simonsolts-bark** [comment]
Thank you for getting back to me; I have created a new issue and linked the logs here: https://github.com/owntone/owntone-server/issues/1545
